### DISCUSSION TITLE
Permission denied for tables as anon

### DISCUSSION THREAD

**Discussion by JacobTheisen** [supabase/supabase#21146] [Questions] [answered]
Ive turned on replication for one of my tables to get real time data in my application. Ive also enabled RLS and made policies for access. 
however when testing the subscription in my application i get the real time data when expected, but i also get a "error 401 unauthorized" and the new, and old data fields are empty.  

<img width="1338" alt="image" src="https://github.com/supabase/supabase/assets/55831903/a5f4e658-a4f7-468d-b1d5-ce7bfb0828af">

<img width="453" alt="image" src="https://github.com/supabase/supabase/assets/55831903/915d9212-265a-496a-9708-9e7bdc318cde">

When i try to view the tables with the anon view in the dashboard i just get permission denied: 
<img width="1349" alt="image" src="https://github.com/supabase/supabase/assets/55831903/d715411b-4f6a-45a0-b2d3-7d45b50ac0a4">

Im fairly new to both supabase and rls so might just be my inexperience, but all help appreciated.

**Comment by GaryAustin1** [comment, marked as answer]
Permission denied for schema is not related to RLS.    It means your Postgres anon role has no access to the public schema for some reason
Do you use Prisma (it modifies grants on schemas)?  Have you modified grants for user roles on schemas?

**Reply by JacobTheisen** [reply]
`Do you use Prisma (it modifies grants on schemas)?`

Yes i am, was not aware prisma was modifying the grants on schemas.

**Comment by GaryAustin1** [comment]
You probably have, but look to troubleshooting here for Prisma and grants:  https://supabase.com/partners/integrations/prisma
